sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -105,6 +105,12 @@ case class CreateViewCommand(
       val tableIdentifier = tableDesc.identifier.copy(database = Option(database))
 
       if (sessionState.catalog.tableExists(tableIdentifier)) {
+        val tableMetadata = sessionState.catalog.getTableMetadata(tableIdentifier)
+        if (tableMetadata.tableType != CatalogTableType.VIEW) {
+          throw new AnalysisException(
```

Comment:
Yeah, a good question. Just did a check using Hive. It is allowed by Hive. 
```
hive> CREATE VIEW IF NOT EXISTS tab1 AS SELECT * FROM t1;
OK
Time taken: 0.678 seconds
```
Let me move this a little bit down.